tests/test_spatialstats.py

```diff
@@ -641,6 +796,107 @@ def test_interp_nd_binning(self):
         # Check a value is returned outside the grid
         assert all(np.isfinite(fun(([-5, 50], [-500,3000], [-2*np.pi,4*np.pi]))))
 
+    def test_two_step_standardization(self):
+        """Test two-step standardization function"""
+
+        ref_dem, diff, mask_glacier, vector_glacier = load_ref_and_diff()
+
+        # Get attributes
+        slope, maximum_curv = xdem.terrain.get_terrain_attribute(ref_dem, attribute=['slope', 'maximum_curvature'])
+
+        # Test this gives the same results as when using the base functions
+        diff_arr = gu.spatial_tools.get_array_and_mask(diff)[0]
+        slope_arr = gu.spatial_tools.get_array_and_mask(slope)[0]
+        maximum_curv_arr = gu.spatial_tools.get_array_and_mask(maximum_curv)[0]
+        stable_mask_arr = ~vector_glacier.create_mask(ref_dem).squeeze()
+
+        # Reproduce the first steps of binning
+        df_binning = xdem.spatialstats.nd_binning(values=diff_arr[stable_mask_arr],
+                                                  list_var=[slope_arr[stable_mask_arr],
+                                                            maximum_curv_arr[stable_mask_arr]],
+                                                  list_var_names=['var1', 'var2'],
+                                                  statistics=[xdem.spatialstats.nmad])
+        unscaled_fun = xdem.spatialstats.interp_nd_binning(df_binning, list_var_names=['var1', 'var2'],
+                                                           statistic='nmad')
+        # The zscore spread should not be one right after binning
+        zscores = diff_arr[stable_mask_arr] / unscaled_fun((slope_arr[stable_mask_arr], maximum_curv_arr[stable_mask_arr]))
+        scale_fac = xdem.spatialstats.nmad(zscores)
+        assert scale_fac != 1
+
+        # Filter with a factor of 5 and the standard deviation and check the function outputs the exact same array
```

Comment:
In the comment you say 5 std, below you use 3 std, and if I recall right, the default value in two_step_standardization was 7. Is there a mix here?

Reply:
I see you have set to 3 below, so only the comment is wrong.

Reply:
Nice catch, it's to test other values than the default (added this detail also)